### PR TITLE
NopTracer is used if tracing is not enabled

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Continuation of discussion https://github.com/eclipse/che/pull/12843#issuecomment-473146651. 
It's bind NopTracer implementation in Guice container in case if tracing is not enabled. No need to check Tracer for null after that.  Removes usage of OptionalTracer.
 
### What issues does this PR fix or reference?
n/A

#### Release Notes
n/A

#### Docs PR
n/A

### CODE REVIEW

**Comment by sleshchenko** [comment]
LGTM
@skabashnyuk Why it is Draft but not PR? Do you have TODO list what should be done before it becomes ready to review?

**Comment by skabashnyuk** [comment]
@sleshchenko  yes. I need more to do more testing with and without tracing enabled.

**Comment by skabashnyuk** [comment]
ci-test

File: core/che-core-tracing-core/src/main/java/org/eclipse/che/core/tracing/NopTracingModule.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import com.google.inject.AbstractModule;
+import io.opentracing.Tracer;
+import io.opentracing.noop.NoopTracerFactory;
+
+/** Guice module that is used if tracing is not enabled. */
+public class NopTracingModule extends AbstractModule {
+
+  @Override
+  protected void configure() {
+    bind(Tracer.class).toInstance(NoopTracerFactory.create());
```
Comment (sleshchenko) [inline core/che-core-tracing-core/src/main/java/org/eclipse/che/core/tracing/NopTracingModule.java]: Would not it better to bind `TracerProvider` here as well?

**Review by sleshchenko — APPROVED** [review]
LGTM

File: core/che-core-tracing-core/src/main/java/org/eclipse/che/core/tracing/NopTracingModule.java (hunk 1 above)
Comment (garagatyi) [reply core/che-core-tracing-core/src/main/java/org/eclipse/che/core/tracing/NopTracingModule.java]: :+1:

**Review by garagatyi — APPROVED** [review]

**Comment by skabashnyuk** [comment]
ci-test

**Review by amisevsk — APPROVED** [review]

**Comment by amisevsk** [comment]
@skabashnyuk I'm +1 to merge this over #12843. I've ported the changes from #12843 onto this branch (diff [here](https://github.com/eclipse/che/compare/NopTracer...amisevsk:issue-12739-b)) and can open a PR once this one is merged.

**Review by metlos — APPROVED** [review]

**Pull request merged**